docs/source/nodes_and_pipelines/pipeline_registry.md

```diff
@@ -0,0 +1,73 @@
+# The pipeline registry
+
+Projects generated using Kedro 0.17.2 or later define their pipelines in `src/<package_name>/pipeline_registry.py`. The `pipeline_registry` module must contain a top-level `register_pipelines()` function that return a mapping from pipeline names to [`Pipeline`](/kedro.pipeline.Pipeline) objects. For example, the pipeline registry in the [Kedro starter for the completed spaceflights tutorial](https://github.com/kedro-org/kedro-starters/tree/main/spaceflights) could define the following `register_pipelines()` function that exposes the data processing pipeline, the data science pipeline, and a third, default pipeline that combines both of the aforementioned pipelines:
+
+```python
+def register_pipelines() -> Dict[str, Pipeline]:
+    """Register the project's pipelines.
+
+    Returns:
+        A mapping from pipeline names to ``Pipeline`` objects.
+    """
+    data_processing_pipeline = dp.create_pipeline()
+    data_science_pipeline = ds.create_pipeline()
+
+    return {
+        "__default__": data_processing_pipeline + data_science_pipeline,
+        "data_processing": data_processing_pipeline,
+        "data_science": data_science_pipeline,
+    }
+```
+
+```{note}
+The order in which you add the pipelines together is not significant (`data_science_pipeline + data_processing_pipeline` would produce the same result), since Kedro automatically detects the data-centric execution order for all the nodes in the resulting pipeline.
+```
+
+## Pipeline autodiscovery
+
+In the above example, you need to update the `register_pipelines()` function whenever you create a pipeline that should be returned as part of the project's pipelines. Since Kedro 0.18.3, you can achieve the same result with less code using `find_pipelines()`[/kedro.framework.project.find_pipelines]`:
+
+```python
+def register_pipelines() -> Dict[str, Pipeline]:
+    """Register the project's pipelines.
+
+    Since Kedro 0.18.3, projects can use the ``find_pipelines`` function
+    to autodiscover pipelines. However, projects that require more fine-
+    grained control can still construct the pipeline mapping without it.
+
+    Returns:
+        A mapping from pipeline names to ``Pipeline`` objects.
+    """
+    pipelines = find_pipelines()
+    pipelines["__default__"] = sum(pipelines.values())
+    return pipelines
+```
+
+Under the hood, the `find_pipelines()` function traverses the `src/<package_name>/pipelines/` directory, returning a mapping from pipeline directory name to [`Pipeline`](/kedro.pipeline.Pipeline) object by:
+
+1. importing the `<package_name>.pipelines.<pipeline_name>` module
+1. calling the `create_pipeline()` function exposed by the `<package_name>.pipelines.<pipeline_name>` module
+1. validating that the constructed object is a [`Pipeline`](/kedro.pipeline.Pipeline)
+
+If any of these steps fail, `find_pipelines()` raises an appropriate warning and skips the current pipeline but continues traversal.
```

Comment:
Maybe add an example of what this warning looks like?

Reply:
The warning depends on which of the above steps fails (or, in the case of "calling the `create_pipeline()` function exposed by the `<package_name>.pipelines.<pipeline_name>` module", which part of the step fails), so it's quite a lot of warnings. I think it would clutter the page to add all of them, and I think the warning messages are fairly self-explanatory if they occur. WDYT?

Reply:
Okay that make sense! I thought maybe all of them include the "Nothing defined therein will be returned by find_pipelines" part, so they know what they look out for. But I'm fine leaving this as is.